### PR TITLE
Update sputility.js \\ SPUserFIeld as hyperlinks, localization of SPBooleanField

### PR DESCRIPTION
**Pull request by szykov**
Two changes made.
1. show SPUserFIeld when use method GetValueString as hyperlinks but not in a plain text. SP normally shows this type of field as hyperlinks on display form.
2. added localization functionality to SPBooleanField. now you can specify how 'Yes' and 'No' in your native language should look like.

### CODE REVIEW

File: dist/sputility.js [hunk 1]
```diff
@@ -1632,8 +1637,59 @@ var SPUtility = (function ($) {
...
+        }
+
+        function getUserId(loginNames, SPUserField) {
+            var d = $.Deferred();
+            var context = new SP.ClientContext.get_current();
+            var arrayLength = loginNames.length;
+            var users = [];
+
+            for (var i = 0; i < arrayLength; i++) {
+                var user = context.get_web().ensureUser(loginNames[i]);
```
Comment (szykov) [inline dist/sputility.js]: her's how i get user info by csom

File: dist/sputility.js [hunk 2]
```diff
@@ -1632,8 +1637,59 @@ var SPUtility = (function ($) {
...
+        var SPUserField = this;
+        var x = getUserId(tmpArray, SPUserField);
+
+        x.done(function (result) {
+            // result is an SP.List because that is what we passed to resolve()!
+            var htmlText = "";
+            for (var i = 0; i < result.users.length; i++) {
+                var user = result.users[i];
+                if (htmlText !== "") { htmlText += "; "; }
+                htmlText += '<a href="/_layouts/15/userdisp.aspx?ID=' + user.get_id().toString() + '&amp;RootFolder=*">' + user.get_title() + '</a>';
```
Comment (szykov) [inline dist/sputility.js]: i retrieve user ids and then i put them into href elements. like this:
`'<a href="/_layouts/15/userdisp.aspx?ID=' + user.get_id().toString() + '&amp;RootFolder=*">' + user.get_title() + '</a>'`

File: dist/sputility.js [hunk 3]
```diff
@@ -1176,7 +1176,7 @@ var SPUtility = (function ($) {
    // overriding the default MakeReadOnly function
    // translate true/false to Yes/No
    SPBooleanField.prototype.MakeReadOnly = function () {
-      return this._makeReadOnly(this.GetValueString());
+      return this._makeReadOnly(this.GetValueString(stringYES, stringNO));
```
Comment (szykov) [inline dist/sputility.js]: her you can specify how 'Yes' and 'No' in your native language.
example:
SPUtility.GetSPField('MyField').MakeReadOnly('Да','Нет');

**Comment by kitmenke** [comment]
Awesome thanks for the PR. I'm going on vacation so I'll take a look when I get back.

**Comment by kitmenke** [comment]
I have made some progress merging this PR. The changes were made to dist\sputility.js which is the output so I've moved them into src\sputility.js otherwise they will get overwritten by the build.

I really like the idea of being able to customize SPBooleanField Yes/No strings. If you had a bunch of them on a single form, passing the strings to every call of MakeReadOnly could get tedious so I'm adding settings for those as well. When I added them I realized that having individual functions for each setting (SetTimeFormat, SetThousandsSeparator, etc) was silly so I'm going to have a single SPUtility.Setup function like so:

```javascript
SPUtility.Setup({                 // DEFAULT SETTINGS:
      'timeFormat': '12HR',      // 12HR or 24HR
      'dateSeparator': '/',      // separates month/day/year with / or .
      'decimalSeparator': '.',   // separates decimal from number
      'thousandsSeparator': ',', // separates thousands in number
      'stringYes': 'Yes',        // Text for when boolean field is True
      'stringNo': 'No'           // Text for when boolean field is False
    });
```

The changes to SPUserField2013 look pretty good. I'm doing some testing now.

Thanks for the hard work @szykov!

**Comment by szykov** [comment]
@kitmenke glad to hear that you liked it. I think you're going in a right direction. having able to configure sputility is a brilliant idea. much cleaner and better. Thanks.

**Comment by kitmenke** [comment]
@szykov You have a twitter?

**Comment by szykov** [comment]
@kitmenke no, where i live twitter isn't popular. if you want to mention me that i helped you with your library on your twitter then calling me something like "random russian guy from Siberia" would be funny and more than enough ^_^

**Pull request merged**